Sources/HummingbirdCompression/CompressedBodyWriter.swift

```diff
@@ -53,10 +57,11 @@ final class CompressedBodyWriter<ParentWriter: ResponseBodyWriter & Sendable>: R
     consuming func finish(_ trailingHeaders: HTTPFields?) async throws {
         // The last buffer must be finished
         if var lastBuffer {
+            guard let compressor else { preconditionFailure("Cannot call finish twice") }
             // keep finishing stream until we don't get a buffer overflow
             while true {
                 do {
-                    try lastBuffer.compressStream(to: &self.window, with: self.compressor, flush: .finish)
+                    try lastBuffer.compressStream(to: &self.window, with: compressor, flush: .finish)
                     try await self.parentWriter.write(self.window)
```

Comment:
If this `write` fails, the allocation isn't freed. Can we use a non-copyable with a deinit to handle this instead? Or a class or something similar.

Reply:
Using AllocatedValue class to hold these.